v2x_ros_driver/src/v2x_radio_client.cpp

```diff
@@ -168,29 +169,138 @@ void V2XRadioClient::process(const std::shared_ptr<const std::vector<uint8_t>>&
         else {
             // TODO lengths greater than 16383 (0x3FFF) are encoded by splitting up the message into discrete chunks, each with its own length
             // marker. It doesn't look like we'll be receiving anything that long
-            std::cerr << "V2XRadioClient::process() : received a message with length field longer than 16383." << std::endl;
+            RCLCPP_WARN_STREAM(logger_, "V2XRadioClient::process() : received a message with length field longer than 16383.");
             continue;
         }
         if (len == -1) { continue; }
         // If the length makes sense bsmPub(fits in the buffer), copy out the message bytes and pass to the Application class
         if ((i + 1 + len + len_bytes) < entry.size()) {
-            // bool found_valid_msg = true;
             if (!IsValidMsgID(std::to_string(msg_id))) { continue; }
+
             size_t start_index = i;
             size_t end_index = i + 2 + len + len_bytes; // includes 2 msgID bytes before message body
-            //this constructor has range [first, last) hence the + 1
+            // this constructor has range [first, last) hence the + 1
             std::vector<uint8_t> msg_vec(entry.begin() + start_index, entry.begin() + end_index);
-            onMessageReceived(msg_vec, msg_id);
-            break;
+
+            // Check if specified MessageFrame size matches actual data size
+            printVector(msg_vec);
+            if (!isValidMsgSize(msg_vec, start_index, end_index, entry))
```

Comment:
It is is a little confusing to step through these nested if-else statements, could a brief comment be added to the branches to help a new user looking at this (i.e. why is a message with a valid message size and an invalid PSID considered a valid message?)

Reply:
Updated